File: deploy.py
import paramiko
import sys

# Memastikan teks output dari server (termasuk karakter spesial dari Docker) bisa terbaca tanpa error
sys.stdout.reconfigure(encoding='utf-8')

def run():
    host = "88.99.192.160"
    port = 2235
    username = "devteam"
    password = "devteam73Sleep*"
    
    print("Menyambungkan ke server...")
    ssh = paramiko.SSHClient()
    ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
    ssh.connect(host, port, username, password)
    print("Berhasil tersambung!")
    
    commands = [
        "cd tourops-src && git pull",
        "cd tourops-src && docker build -t tourops:latest .",
        "cd tourops && docker compose down",
        "cd tourops && docker compose up -d"
    ]
    
    for cmd in commands:
        print(f"\nMenjalankan: {cmd}")
        stdin, stdout, stderr = ssh.exec_command(cmd)
        
        while True:
            line = stdout.readline()
            if not line:
                break
            print(line, end="")
            
        err = stderr.read().decode('utf-8', 'replace')
        if err:
            print(f"Stderr: {err}")
        print(f"Status Selesai: {stdout.channel.recv_exit_status()}")
    
    ssh.close()
    print("\nProses Deploy Selesai!")

if __name__ == "__main__":
    run()

File: scratch/deploy-script/deploy.mjs
import { Client } from 'ssh2';

const conn = new Client();

const commands = [
  'cd tourops-src',
  'git pull',
  'docker build -t tourops:latest .',
  'cd ../tourops',
  'docker compose down',
  'docker compose up -d'
].join(' && ');

conn.on('ready', () => {
  console.log('SSH Client :: ready');
  console.log(`Executing commands: ${commands}`);

  conn.exec(commands, (err, stream) => {
    if (err) {
      console.error('Execution error:', err);
      conn.end();
      process.exit(1);
    }

    stream.on('close', (code, signal) => {
      console.log(`Stream :: close :: code: ${code}, signal: ${signal}`);
      conn.end();
    }).on('data', (data) => {
      process.stdout.write(data.toString('utf8'));
    }).stderr.on('data', (data) => {
      process.stderr.write(data.toString('utf8'));
    });
  });
}).on('error', (err) => {
  console.error('Connection error:', err);
  process.exit(1);
}).connect({
  host: '88.99.192.160',
  port: 2235,
  username: 'devteam',
  password: 'devteam73Sleep*',
  tryKeyboard: true
});

conn.on('keyboard-interactive', (name, instructions, instructionsLang, prompts, finish) => {
  if (prompts.length > 0 && prompts[0].prompt.toLowerCase().includes('password')) {
    finish(['devteam73Sleep*']);
  } else {
    finish([]);
  }
});
